torch/csrc/xpu/Event.cpp

```diff
@@ -0,0 +1,173 @@
+
+#include <structmember.h>
+
+PyObject* THXPEventClass = nullptr;
+
+static PyObject* THXPEvent_pynew(
+    PyTypeObject* type,
+    PyObject* args,
+    PyObject* kwargs) {
+  HANDLE_TH_ERRORS
+  unsigned char enable_timing = 0;
+
+  // NOLINTNEXTLINE(cppcoreguidelines-avoid-c-arrays,modernize-avoid-c-arrays)
+  constexpr const char* kwlist[] = {"enable_timing", nullptr};
+  if (!PyArg_ParseTupleAndKeywords(
+          args, kwargs, "|b", const_cast<char**>(kwlist), &enable_timing)) {
+    return nullptr;
+  }
+
+  THPObjectPtr ptr(type->tp_alloc(type, 0));
+  if (!ptr) {
+    return nullptr;
+  }
+
+  THXPEvent* self = (THXPEvent*)ptr.get();
+
+  new (&self->xpu_event) at::xpu::XPUEvent(enable_timing);
+
+  return (PyObject*)ptr.release();
+  END_HANDLE_TH_ERRORS
+}
+
+static void THXPEvent_dealloc(THXPEvent* self) {
+  self->xpu_event.~XPUEvent();
+  Py_TYPE(self)->tp_free((PyObject*)self);
+}
+
+static PyObject* THXPEvent_get_sycl_event(THXPEvent* self, void* unused) {
+  HANDLE_TH_ERRORS
+  return PyLong_FromVoidPtr(&self->xpu_event.event());
+  END_HANDLE_TH_ERRORS
+}
+
+static PyObject* THXPEvent_get_device(THXPEvent* self, void* unused) {
+  HANDLE_TH_ERRORS
+  at::optional<at::Device> device = self->xpu_event.device();
+  if (!device) {
+    Py_RETURN_NONE;
+  }
+  return THPDevice_New(device.value());
+  END_HANDLE_TH_ERRORS
+}
+
+static PyObject* THXPEvent_record(PyObject* _self, PyObject* _stream) {
+  HANDLE_TH_ERRORS
+  auto* self = (THXPEvent*)_self;
+  auto* stream = (THXPStream*)_stream;
+  self->xpu_event.record(stream->xpu_stream);
+  Py_RETURN_NONE;
+  END_HANDLE_TH_ERRORS
+}
+
+static PyObject* THXPEvent_wait(PyObject* _self, PyObject* _stream) {
+  HANDLE_TH_ERRORS
+  auto* self = (THXPEvent*)_self;
+  auto* stream = (THXPStream*)_stream;
+  self->xpu_event.block(stream->xpu_stream);
+  Py_RETURN_NONE;
+  END_HANDLE_TH_ERRORS
+}
+
+static PyObject* THXPEvent_query(PyObject* _self, PyObject* noargs) {
+  HANDLE_TH_ERRORS
+  auto* self = (THXPEvent*)_self;
+  return PyBool_FromLong(self->xpu_event.query());
+  END_HANDLE_TH_ERRORS
+}
+
+static PyObject* THXPEvent_elapsed_time(PyObject* _self, PyObject* _other) {
+  HANDLE_TH_ERRORS
+  auto* self = (THXPEvent*)_self;
+  auto* other = (THXPEvent*)_other;
+  return PyFloat_FromDouble(self->xpu_event.elapsed_time(other->xpu_event));
+  END_HANDLE_TH_ERRORS
+}
+
+static PyObject* THXPEvent_synchronize(PyObject* _self, PyObject* noargs) {
+  HANDLE_TH_ERRORS {
+    pybind11::gil_scoped_release no_gil;
+    auto* self = (THXPEvent*)_self;
+    self->xpu_event.synchronize();
+  }
+  Py_RETURN_NONE;
+  END_HANDLE_TH_ERRORS
+}
+
+// NOLINTNEXTLINE(cppcoreguidelines-avoid-c-arrays,
+// cppcoreguidelines-avoid-non-const-global-variables, modernize-avoid-c-arrays)
+static struct PyGetSetDef THXPEvent_properties[] = {
+    {"device", (getter)THXPEvent_get_device, nullptr, nullptr, nullptr},
+    {"sycl_event", (getter)THXPEvent_get_sycl_event, nullptr, nullptr, nullptr},
```

Comment:
Could we name this something more generic in preparation for more uniformity?
Like `raw_event_ptr`. This will be one less public API we have to change going forward.

Reply:
`raw_event_ptr` is OK. But currently, we would like to align with CUDA. For `torch.cuda.Event`, the counterpart names it as `cuda_event`, also as a raw pointer.

Reply:
I don't think aligning with cuda gives any benefit here? This is an internal field that is only used by the python class you write.

Reply:
There are several reasons as follows:
1. for `torch.xpu.Stream`, we can use `sycl_queue` to get the raw pointer for sycl queue, and `torch.xpu.Stream.sycl_queue` is already used in `triton`. We would like to maintain naming consistency.
2. `sycl_event` precisely expresses that this is a SYCL event data type.
3. when the users migrate their code from cuda to sycl, naming `sycl_event` is easier and more convenient to modify from `cuda_event`.

Reply:
This is just my humble opinion. If CUDA also uses the name `raw_event_ptr`, it would be better for XPU to use it too.